refl1d/magnetism.py

```diff
@@ -505,21 +516,25 @@ def to_dict(self):
 
     def profile(self, Pz, thickness):
         z = hstack((0, cumsum(asarray([v.value for v in self.dz], 'd'))))
+        if z[-1] == 0:
+            z[-1] = 1
         z *= thickness/z[-1]
 
         rhoM = hstack((0, cumsum(asarray([v.value for v in self.drhoM], 'd'))))
         # AJC added since without the line below FreeMagnetismInterface
         # does not initialise propoerly - fixes strange behaviour at drho=0 on end point
         if rhoM[-1] == 0:
             rhoM[-1] = 1
-        if rhoM[-1] > 0:
-            rhoM *= 1/rhoM[-1]
+
+        rhoM *= 1/rhoM[-1]
```

Comment:
The pre-existing variable names are misleading: `rhoM` and `thetaM` should be `rhoM_fraction` and `thetaM_fraction` or some such since the represent the left/right blend portion.

Reply:
Ok, I am happy to go through and change instances of `rhoM` and `thetaM` to `rhoM_fraction` and `thetaM_fraction` respectively.